src/abs.rs

```diff
@@ -19,19 +50,118 @@ use super::{PathArc, PathDir, PathFile, Result};
 pub struct PathAbs(pub(crate) PathArc);
 
 impl PathAbs {
-    /// Instantiate a new `PathAbs`. The path must exist or `io::Error` will be returned.
+    /// Construct an absolute path from an arbitrary (absolute or relative) one.
+    ///
+    /// This is different from [`canonicalize`] in that it _preserves_ symlinks
+    /// and the destination may or may not exist.
+    ///
+    /// This function will:
+    /// - Resolve relative paths against the current working directory.
+    /// - Strip any `.` components (`/a/./c` -> `/a/c`)
+    /// - Resolve `..` _semantically_ (not using the file system). So, `a/b/c/../d => a/b/d` will
+    ///   _always_ be true regardless of symlinks. If you want symlinks correctly resolved, use
+    ///   `canonicalize()` instead.
+    ///
+    /// > On windows, this will sometimes call `canonicalize()` on the first component to guarantee
+    /// > it is the correct canonicalized prefix. For paths starting with root it also has to get
+    /// > the [`current_dir`]
+    ///
+    /// > On linux, the only syscall this will make is to get the [`current_dir`] for relative
+    /// > paths.
+    ///
+    /// [`canonicalize`]: struct.PathAbs.html#method.canonicalize
+    /// [`current_dir`]: fn.current_dir.html
     ///
     /// # Examples
+    ///
     /// ```rust
     /// use path_abs::PathAbs;
     ///
     /// # fn try_main() -> ::std::io::Result<()> {
     /// let lib = PathAbs::new("src/lib.rs")?;
+    ///
+    /// assert_eq!(lib.is_absolute(), true);
     /// # Ok(()) } fn main() { try_main().unwrap() }
     /// ```
     pub fn new<P: AsRef<Path>>(path: P) -> Result<PathAbs> {
-        let arc = PathArc::new(path);
-        arc.absolute()
+        let path = PathArc::new(path);
```

Comment:
This is copy/paste from our past review, right?

Also, was there really no `new` method on `PathAbs`??? I can't believe I didn't include that!

Reply:
Yes, this is just moving around the code introduced in a previous review.

There *was* a `new` method, but it just called `PathArc::absolute()`.